qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/ParameterizedFullClusterRestartTestCase.java

```diff
@@ -78,6 +80,16 @@ public static org.elasticsearch.Version getOldClusterVersion() {
         return org.elasticsearch.Version.fromString(OLD_CLUSTER_VERSION.toString());
     }
 
+    public static IndexVersion getOldClusterIndexVersion() {
+        var version = getOldClusterVersion();
+        if (version.equals(org.elasticsearch.Version.CURRENT)) {
+            return IndexVersion.current();
+        } else {
+            assertThat("Index version needs to be added to restart test parameters", version, lessThan(org.elasticsearch.Version.V_8_10_0));
```

Comment:
This is a time bomb, failing once we branch 8.10 right?

Reply:
Yes (at least, it will fail when the old version is 8.10.0). On second thoughts, this should probably be an `assume`. There's two ways we could handle this at this point:

1. Change it to an `assume`, so the test doesn't fail outright, it just stops working once the code is branched
2. Leave it as an incentive to get the work done properly before 8.10

Reply:
I'd rather have the latter. Otherwise this just gets silently ignored once we branch 8.10.

Reply:
OK, marking #97200 as a 8.10 blocker